unstructured/ingest/cli/interfaces.py

```diff
@@ -397,13 +397,25 @@ def get_cli_options() -> t.List[click.Option]:
 class CliEmbeddingConfig(EmbeddingConfig, CliMixin):
     @staticmethod
     def get_cli_options() -> t.List[click.Option]:
+        from unstructured.embed import EMBEDDING_PROVIDER_TO_CLASS_MAP
+
         options = [
+            click.Option(
+                ["--embedding-provider"],
+                help="Type of the embedding class to be used. Can be one of: "
+                f"{list(EMBEDDING_PROVIDER_TO_CLASS_MAP)}",
```

Comment:
leverage the [choice](https://click.palletsprojects.com/en/8.1.x/options/#choice-options) click option to allow click to throw an appropriate error if they don't pass in something that is allowed. 
```python
type=click.Choice(list(EMBEDDING_PROVIDER_TO_CLASS_MAP))
```

Reply:
This is cool.

Offloaded raising to Click, and removed the raise logic from EmbeddingConfig: 
[check provider error in cli via Click, instead of in EmbeddingConfig](https://github.com/Unstructured-IO/unstructured/pull/1918/commits/a8b95fdf84f9428f6595e27c42475ceb2c44d75b)

Let me know if we still would like to keep the try / except within EmbeddingConfig